cpp/src/svm/results.h

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuda_utils.h>
+#include <math.h>
+#include <iostream>
+#include <limits>
+#include <memory>
+
+#include <cub/device/device_select.cuh>
+#include "common/cumlHandle.hpp"
+#include "common/cuml_allocator.hpp"
+#include "common/device_buffer.hpp"
+#include "common/host_buffer.hpp"
+#include "kernelcache.h"
+#include "linalg/binary_op.h"
+#include "linalg/init.h"
+#include "linalg/map_then_reduce.h"
+#include "linalg/unary_op.h"
+#include "ws_util.h"
+
+namespace ML {
+namespace SVM {
+
+template <typename math_t, typename Lambda>
+__global__ void set_flag(bool *flag, const math_t *alpha, int n, Lambda op) {
```

Comment:
I think you can use either binary_op.h or unary_op.h for this operation. You don't need a separate kernel.